server/task_specs/feature-match.py

```diff
@@ -40,22 +40,65 @@ def compute_assignments(list1, list2, metric=None):
     assign1, assign2 = (
         optimize.linear_sum_assignment(cost_matrix))
 
+    assign1 = assign1.astype(np.uint64)
+    assign2 = assign2.astype(np.uint64)
+
+    n1 = len(list1)
+    n2 = len(list2)
+    n_min = min(n1, n2)
+
+    # no more than *one* of these branches may be executed
+    left_over = None
+    if n1 != n_min:
+        left_over = np.empty((n1,), dtype=np.bool)
+        target = assign1
+
+    if n2 != n_min:
```

Comment:
Can you use `elif` here? If the comment above is accurate, `elif` would enforce the "at most one branch" constraint.